consensus/blocks.go

```diff
@@ -107,13 +100,16 @@ func (s *State) validHeader(b Block) (err error) {
 		return EarlyTimestampErr
 	}
 
-	// Check that the block is not too far in the future.
-	skew := b.Timestamp - Timestamp(time.Now().Unix())
+	// Check that the block is not too far in the future. An external process
+	// will need to be responsible for resubmitting the block once it is no
+	// longer in the future.
+	skew := int(b.Timestamp) - int(Timestamp(time.Now().Unix()))
```

Comment:
subtraction.

timestamp - (slightly bigger timestamp) = super big timestamp, because timestamps are unsigned. but casting it to an int means that you get a negative timestamp, which is acceptable.

except (derp), this comparison still has problems because you can submit a super huge timestamp and it'll evaluate to being really far in the past. So the timestamp checking code needs some reworking.